lib/jxl/modular/transform/squeeze.cc

```diff
@@ -12,13 +12,155 @@
 #include "lib/jxl/common.h"
 #include "lib/jxl/modular/modular_image.h"
 #include "lib/jxl/modular/transform/transform.h"
-
+#undef HWY_TARGET_INCLUDE
+#define HWY_TARGET_INCLUDE "lib/jxl/modular/transform/squeeze.cc"
+#include <hwy/foreach_target.h>
+#include <hwy/highway.h>
+HWY_BEFORE_NAMESPACE();
 namespace jxl {
+namespace HWY_NAMESPACE {
+
+using hwy::HWY_NAMESPACE::ShiftLeft;
+using hwy::HWY_NAMESPACE::ShiftRight;
+
+#if HWY_TARGET != HWY_SCALAR
+
+template <size_t count>
+JXL_INLINE void fast_unsqueeze(const pixel_type *JXL_RESTRICT p_residual,
+                               const pixel_type *JXL_RESTRICT p_avg,
+                               const pixel_type *JXL_RESTRICT p_navg,
+                               const pixel_type *p_pout,
+                               pixel_type *JXL_RESTRICT p_out,
+                               pixel_type *p_nout) {
+  const HWY_CAPPED(pixel_type, count) d;
+  const size_t N = Lanes(d);
+  auto onethird = Set(d, 0x55555556);
+  for (size_t x = 0; x < count; x += N) {
+    auto avg = Load(d, p_avg + x);
+    auto next_avg = Load(d, p_navg + x);
+    auto top = Load(d, p_pout + x);
+    // Equivalent to SmoothTendency(top,avg,next_avg), but without branches
+    auto Ba = top - avg;
+    auto an = avg - next_avg;
+    auto nonmono = Ba ^ an;
+    auto absBa = Abs(Ba);
+    auto absan = Abs(an);
+    auto absBn = Abs(top - next_avg);
+    auto a3e = BitCast(d, ShiftRight<32>(MulEven(absBa, onethird)));
```

Comment:
A comment here would be _very_ helpful (or factoring out to a DivideBy3)

Reply:
Added a comment.